crates/bevy_ui/src/experimental/ghost_hierarchy.rs

```diff
@@ -14,10 +15,30 @@ use crate::Node;
 /// The UI systems will traverse past these and treat their first non-ghost descendants as direct children of their first non-ghost ancestor.
 ///
 /// Any components necessary for transform and visibility propagation will be added automatically.
-#[derive(Component, Default, Debug, Copy, Clone, Reflect)]
+///
+/// Instances of this type cannot be constructed unless the `ghost_nodes` feature is enabled.
+#[derive(Component, Debug, Copy, Clone, Reflect)]
+#[cfg_attr(feature = "ghost_nodes", derive(Default))]
 #[reflect(Component, Debug)]
 #[require(Visibility, Transform)]
-pub struct GhostNode;
+pub struct GhostNode {
+    // This is a workaround to ensure that GhostNode is only constructable when the appropriate feature flag is enabled
+    #[reflect(ignore)]
+    unconstructable: PhantomData<()>, // Spooky!
+}
```

Comment:
```suggestion
pub struct GhostNode(
    // This is a workaround to ensure that GhostNode is only constructable when the appropriate feature flag is enabled
    #[cfg(not(feature = "ghost_nodes"))]
    #[reflect(ignore)]
    unconstructable: PhantomData<()>, // Spooky!
);
```

Switching to a tuple-struct instead and using the `not` feature gating allows this to be constructed as before with the feature enabled:

```rust
// With feature "ghost_nodes"
let node = GhostNode;
```

Reply:
Cursed. Okay, let's try that!